File: j-LIB-core/src/main/java/io/github/dot166/jLib/LIBAboutActivity.java
package io.github.dot166.jLib;

import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.util.Log;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.content.res.AppCompatResources;

import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

import io.github.dot166.jLib.app.jAboutActivity;

public class LIBAboutActivity extends jAboutActivity {

    @NonNull
    @Override
    public Intent versionIntent(@NonNull Context context) {
        return new Intent(context, LIBTestActivity.class);
    }

    @NonNull
    @Override
    public Drawable getActivityIcon(@NonNull Context context) {
        return Objects.requireNonNull(AppCompatResources.getDrawable(context, R.mipmap.ic_launcher_j));
    }

    @NonNull
    @Override
    public String getAppLabel(@NonNull Context context) {
        return getString(R.string.jlib);
    }

    @NonNull
    @Override
    public String versionName(@NonNull Context context) {
        return BuildConfig.LIBVersion;
    }

    @Override
    public boolean hideAcknowledgements(@NonNull Context context) {
        return true;
    }

    @NonNull
    @Override
    public List<jAboutActivity.Contributor> product() {
        return new ArrayList<>() {{
            add(new Contributor("._______166", Role.LeadDev, "https://avatars.githubusercontent.com/u/62702353", "https://github.com/dot166"));
            add(new Contributor("bh916", Role.Dev, "https://avatars.githubusercontent.com/u/138221251", "https://github.com/bh196"));
        }};
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getSupportActionBar() != null) {
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
            getSupportActionBar().setHomeActionContentDescription(androidx.appcompat.R.string.abc_action_bar_up_description);
        } else {
            Log.e("ActionBar2", "no actionbar found");
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                getOnBackPressedDispatcher().onBackPressed();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    public enum Role implements Roles {
        LeadDev(R.string.leaddev),
        Dev(R.string.dev);

        private final int descriptionResId;

        Role(int descriptionResId) {
            this.descriptionResId = descriptionResId;
        }

        public int getDescriptionResId() {
            return descriptionResId;
        }

        @Override
        public int descriptionResId() {
            return this.descriptionResId;
        }
    }

}

File: j-LIB-core/src/main/java/io/github/dot166/jlib/LIBAboutActivity.kt
package io.github.dot166.jlib

import android.content.Context
import android.content.Intent
import android.graphics.drawable.Drawable
import androidx.appcompat.content.res.AppCompatResources
import io.github.dot166.jlib.app.jAboutActivity
import io.github.dot166.jlib.utils.VersionUtils.libVersion
import java.util.Objects

class LIBAboutActivity : jAboutActivity() {
    override fun versionIntent(context: Context): Intent {
        return Intent(context, LIBTestActivity::class.java)
    }

    override fun getAppIcon(context: Context): Drawable {
        return Objects.requireNonNull<Drawable>(
            AppCompatResources.getDrawable(
                context,
                R.mipmap.ic_launcher_j
            )
        )
    }

    override fun getAppLabel(context: Context): String {
        return getString(R.string.jlib)
    }

    override fun versionName(context: Context): String {
        return libVersion
    }

    override fun product(): List<Contributor> {
        return object : ArrayList<Contributor>() {
            init {
                add(
                    Contributor(
                        "._______166",
                        Role.Maintainer,
                        "https://avatars.githubusercontent.com/u/62702353",
                        "https://github.com/dot166"
                    )
                )
                add(
                    Contributor(
                        "bh916",
                        Role.Contributor,
                        "https://avatars.githubusercontent.com/u/138221251",
                        "https://github.com/bh196"
                    )
                )
            }
        }
    }

    enum class Role(val descriptionResId: Int) : Roles {
        Maintainer(R.string.maintainer),
        Contributor(R.string.contributor);

        override fun descriptionResId(): Int {
            return this.descriptionResId
        }
    }
}
